crates/evm/src/evm/db_commit.rs

```diff
@@ -51,8 +51,8 @@ impl<'a, C: sov_modules_api::Context> DatabaseCommit for EvmDb<'a, C> {
             let account_info = account.info;
 
             if let Some(ref code) = account_info.code {
-                if !code.is_empty() && !self.code.exists(&account_info.code_hash, self.working_set)
-                {
+                if !code.is_empty() {
+                    // TODO: would be good to have a contains_key method on the StateMap that would be optimized, so we can check the hash before storing the code
                     self.code
                         .set(&account_info.code_hash, code, self.working_set);
```

Comment:
I wonder if we set the code state map to the same code again in a batch proof, do we include that in the state diff?

Reply:
No, it's gonna be only 1, because we keep only the last (k, v) pair.

Reply:
no but imagine there is two different proofs 

proof 1: sequencer commitmenti in L1 #12
proof 2: sequencer commitment in L1 #15 

in proof one I deploy contract x 

state diff will have the code (as it should be)

in proof two I also deploy contract x
the value in the state tree does not change but will it be in the state diff????

Reply:
it's ok. sorry for the misunderstanding. we have the if check now